lib/onelogin/ruby-saml/logoutresponse.rb

```diff
@@ -69,46 +103,202 @@ def status_code
         end
       end
 
+      # @return [String] Gets the StatusMessage from a SAML Response.
+      #
+      def status_message
+        @status_message ||= begin
+          node = REXML::XPath.first(document, "/p:LogoutResponse/p:Status/p:StatusMessage", { "p" => PROTOCOL, "a" => ASSERTION })
+          node.text if node
+        end
+      end
+
+      # Gets the expected current_url
+      # (Right now we read this url from the Sinle Logout Service of the Settings)
+      # TODO: Calculate the real current_url and use it.
+      # @return [String] The current url
+      #
+      def current_url
+        @current_url ||= begin
+          unless self.settings.nil? or self.settings.single_logout_service_url.nil?
```

Comment:
I'd also recommend using `if` instead of `unless` here to make this more readable:
```ruby
if settings && settings.single_logout_service_url
  ...
end
```
I'd also avoid using `self` unless there's a method that could be confused with a passed variable with the same name, see how on the previous methods we don't use `self.document` to get the values for that accessor